common/shield_exceptions.cc

```diff
@@ -52,6 +52,29 @@ bool IsWhitelistedReferrer(const GURL& firstPartyOrigin,
   // Note that there's already an exception for TLD+1, so don't add those here.
   // Check with the security team before adding exceptions.
 
+  // https://github.com/brave/browser-laptop/issues/5861
+  // The below patterns are done to only allow the specific request
+  // pattern, of reddit -> redditmedia -> embedly -> imgur.
+  static auto redditPtrn = URLPattern(URLPattern::SCHEME_ALL, "https://www.reddit.com/*");
+  static std::vector<URLPattern> reddit_embed_patterns({
+    redditPtrn,
+    URLPattern(URLPattern::SCHEME_ALL, "https://www.redditmedia.com/*"),
+    URLPattern(URLPattern::SCHEME_ALL, "https://cdn.embedly.com/*"),
+    URLPattern(URLPattern::SCHEME_ALL, "https://imgur.com/*")
```

Comment:
can we make all of these URLPattern::SCHEME_HTTPS instead of SCHEME_ALL? for instance referrer should never be sent to an HTTP site on an HTTPS site anyway

Reply:
@diracdeltas fixed!